core/execution/future/market.go

```diff
@@ -1559,13 +1559,35 @@ func (m *Market) leaveAuction(ctx context.Context, now time.Time) {
 		m.risk.GetRiskFactors().Long)
 
 	if wasOpeningAuction && (m.getCurrentMarkPrice().IsZero()) {
-		m.markPriceCalculator.overridePrice(m.lastTradedPrice)
+		t := m.timeService.GetTimeNow().UnixNano()
+		if wasOpeningAuction {
+			// if we're in opening auction then orderbook-based price will have a duration of 0 (the price and funding period start coincide) hence we first calculate the orderbook price at current time and then calculate the composite price assuming time has moved by the smallest possible increment
+			m.markPriceCalculator.CalculateBookMarkPriceAtTimeT(m.tradableInstrument.MarginCalculator.ScalingFactors.InitialMargin, m.mkt.LinearSlippageFactor, m.risk.GetRiskFactors().Short, m.risk.GetRiskFactors().Long, t, m.matching)
```

Comment:
shouldn't we do this thing anyways before line 1552 regardless if we get 0 or nil?

Reply:
I think that's fine too